### PR TITLE
Implement ClusterSharding.shardRegion for exposing Akka Cluster Sharding region ActorRef

### PR DESCRIPTION
**Pull request by dfakhritdinov**
The method will be used to emulate recovery after rebalancing

### CODE REVIEW

**Comment by dfakhritdinov** [comment]
Please ignore **Codacy Static Code Analysis** failure cos it cannot differentiate `Option` and CE `Ref` :D 

Screenshot 2024-05-22 at 18 17 23

File: cluster-sharding/src/main/scala/com/evolutiongaming/akkaeffect/cluster/sharding/ClusterSharding.scala [hunk 1]
```diff
@@ -62,7 +67,7 @@ object ClusterSharding {
     val actorRefOf = ActorRefOf.fromActorRefFactory[F](actorSystem)
     val terminated = Terminated[F](actorRefOf)
 
-    def shardRegion(actorRef: => ActorRef) =
+    def createShardRegion(actorRef: => ActorRef) =
```
Comment (mr-git) [inline cluster-sharding/src/main/scala/com/evolutiongaming/akkaeffect/cluster/sharding/ClusterSharding.scala:70]: isn't this the public API change?

shouldn't we deprecate with hit about new API the original?:
```scala
def shardRegion(actorRef: => ActorRef): Resource[F, ActorRef]
```

Comment (dfakhritdinov) [reply cluster-sharding/src/main/scala/com/evolutiongaming/akkaeffect/cluster/sharding/ClusterSharding.scala:70]: It looks tricky but its not an API change, let me explain: I've added new API method
```
def shardRegion(typeName: TypeName): F[ActorRef]
```

and implement it on line #135. Unfortunately there was already a private method with same name and they clash in same scope. Thus I renamed private method to `createShardRegion`. Btw, could it be that MiMa by mistake took it as public API change? I have no idea how MiMa works and does it parses the code and builds ADT or just uses regex magic 🤔

File: version.sbt [hunk 2]
```diff
@@ -1 +1 @@
-ThisBuild / version := "4.0.8-SNAPSHOT"
+ThisBuild / version := "4.1.0"
```
[the remark this answers is not in the document]
Comment (dfakhritdinov) [reply version.sbt:1]: that is my plan because I need this change for tests in some Akka related stuff

**Review by mr-git — APPROVED** [review]

**Pull request merged**